#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix missing __GI___pwrite64.
# DP: Author: Phil Blundell <pb@debian.org>
# DP: Upstream status: Pending
# DP: Status Details: Submitted Jul 29, 2003

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.

2003-07-29  Philip Blundell  <philb@gnu.org>

	* sysdeps/unix/sysv/linux/alpha/syscalls.list (pread): Syscall
	name is pread, not pread64.
	(pwrite): Likewise vs pwrite64.

--- sysdeps/unix/sysv/linux/alpha/syscalls.list	24 Jun 2003 16:26:34 -0000	1.59
+++ sysdeps/unix/sysv/linux/alpha/syscalls.list	29 Jul 2003 19:23:02 -0000
@@ -22,8 +22,8 @@
 llseek		EXTRA	lseek		C:3	__libc_lseek	__lseek lseek __libc_lseek64 __llseek llseek __lseek64 lseek64
 lseek		llseek	-
 posix_fadvise64	-	fadvise64	4	posix_fadvise64	posix_fadvise
-pread		-	pread64		C:4	__libc_pread	__libc_pread64 __pread pread __pread64 pread64
-pwrite		-	pwrite64		C:4	__libc_pwrite	__libc_pwrite64 __pwrite pwrite __pwrite64 pwrite64
+pread		-	pread		C:4	__libc_pread	__libc_pread64 __pread pread __pread64 pread64
+pwrite		-	pwrite		C:4	__libc_pwrite	__libc_pwrite64 __pwrite pwrite __pwrite64 pwrite64
 fstatfs		-	fstatfs		i:ip	__fstatfs	fstatfs __fstatfs64 fstatfs64
 statfs		-	statfs		i:sp	__statfs	statfs statfs64
 getrlimit	-	getrlimit	2	__getrlimit	getrlimit getrlimit64
